### PR TITLE
[FIX] owcorrespondence: Handle variables with one value

### PR DESCRIPTION
**Pull request by jerneju**
##### Issues
1) IndexError: index 1 is out of bounds for axis 1 with size 1
2) SVD did not converge
3) Must specify at least one of rect, xRange, or yRange. (gave rect=<class 'NoneType'>)

Check that the widget does not crash when domain has a discrete variable with only one value.

##### Description of changes
Discrete variables with only one value are not taken into account.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/220489676/

File: Orange/widgets/unsupervised/owcorrespondence.py [hunk 1]
```diff
@@ -108,7 +107,7 @@ def set_data(self, data):
         self.data = data
         if data is not None:
             self.varlist[:] = [var for var in data.domain.variables
-                               if var.is_discrete]
+                               if var.is_discrete and len(var.values) > 1]
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owcorrespondence.py]: This only filters out variables with less than 2 values.
A variable can have 3 values, but after selecting a subset only instances with 1 will remain.
This widget will still crash.

**Comment by lanzagar** [comment]
When a PR is ready for review and merging, please try to give it a good name (that can appear in a Changelog) - the python error from the bug is not the most informative.
E.g. this could be:
[FIX] owcorrespondence: Handle variables with one value

File: Orange/widgets/unsupervised/owcorrespondence.py (hunk 1 above)
Comment (jerneju) [inline Orange/widgets/unsupervised/owcorrespondence.py]: Yes, I get "SVD did not converge" error.

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/204513970/

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/196882642/

**Pull request merged**